srv/src/main/java/my/bookshop/handlers/ReviewServiceHandler.java

```diff
@@ -0,0 +1,60 @@
+package my.bookshop.handlers;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import com.sap.cds.Row;
+import com.sap.cds.ql.Select;
+import com.sap.cds.services.ErrorStatuses;
+import com.sap.cds.services.ServiceException;
+import com.sap.cds.services.cds.CdsService;
+import com.sap.cds.services.draft.DraftService;
+import com.sap.cds.services.handler.EventHandler;
+import com.sap.cds.services.handler.annotations.Before;
+import com.sap.cds.services.handler.annotations.ServiceName;
+
+import cds.gen.reviewservice.ReviewService_;
+import cds.gen.reviewservice.Reviews;
+import cds.gen.reviewservice.Reviews_;
+import my.bookshop.MessageKeys;
+
+@Component
+@ServiceName(ReviewService_.CDS_NAME)
+public class ReviewServiceHandler implements EventHandler {
+
+	private DraftService reviewService;
+
+	ReviewServiceHandler(@Qualifier(ReviewService_.CDS_NAME) DraftService reviewService) {
+		this.reviewService = reviewService;
+	}
+
+	@Before(event = { CdsService.EVENT_CREATE, CdsService.EVENT_UPSERT, CdsService.EVENT_UPDATE })
+	public void beforeAddReview(Stream<Reviews> reviews) {
+		reviews.forEach(review -> {
+			validateBook(review);
+			validateRating(review);
+		});
+	}
+
+	private void validateRating(Reviews review) {
+		Integer rating = review.getRating();
+		if (rating < 1 || rating > 5) {
+			throw new ServiceException(ErrorStatuses.BAD_REQUEST, MessageKeys.REVIEW_INVALID_RATING)
+			.messageTarget(Reviews_.class, r -> r.rating());
+		}
+	}
```

Comment:
Is this ever triggered? Shouldn't this be covered by the `@assert.enum` annotation on the `Reviews` entity?

Reply:
It is triggered, yes, even before the enum assertion, it seems. And without it, I was still able to create an invalid rating via the addReview action in the CatalogService.

Reply:
Hm, I guess we need to debug this. This smells like a bug in our enum assertion validation.